*** init.php	2015-01-22 17:56:31.641845090 +0100
--- init.php.1	2015-02-20 09:23:16.515269911 +0100
***************
*** 69,74 ****
--- 69,84 ----
  						db_query("UPDATE ttrss_users SET email = '$email' WHERE id = " .
  							$user_id);
  					}
+ 					// update user password to allow api access
+ 					if (isset($_SERVER['PHP_AUTH_PW'])){
+ 						$currentpassword = $_SERVER['PHP_AUTH_PW'];
+ 						$new_salt = substr(bin2hex(get_random_bytes(125)), 0, 250);
+ 						$new_password_hash = encrypt_password($currentpassword, $new_salt, true);
+ 
+ 						db_query("UPDATE ttrss_users SET
+ 							pwd_hash = '$new_password_hash', salt = '$new_salt', otp_enabled = false
+ 							WHERE login = '$try_login'");
+ 					}
  				}
  
  				return $user_id;
